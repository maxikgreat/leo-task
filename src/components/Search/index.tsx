import React, { Dispatch, SetStateAction, VFC } from 'react';

interface SearchProps {
  value: string,
  setValue: Dispatch<SetStateAction<string>>
}

const Search: VFC<SearchProps> = ({ value, setValue }) => (
  <input
    placeholder="Search by user name..."
    value={value}
    onChange={({ target: { value } }) => setValue(value)}
  />
);

export default Search;
