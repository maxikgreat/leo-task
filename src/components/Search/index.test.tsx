import { render, fireEvent } from '@testing-library/react';

import Search from '.';
import { useState, VFC } from 'react';

describe('Search component', () => {
  const placeholder = 'Search by user name...';
  it('should render correctly and match snapshot', () => {
    const value = 'Example value';

    const elem = render(<Search
      value={value}
      setValue={jest.fn()}
    />);

    expect((elem.getByPlaceholderText(placeholder) as HTMLInputElement).value)
      .toBe(value);
    expect(elem.getByPlaceholderText(placeholder)).toMatchSnapshot();
  });

  it('should change value correctly', () => {
    const newValue = 'New value';
    const StateWrapper: VFC = () => {
      const [search, setSearch] = useState('');

      return (
        <Search value={search} setValue={setSearch} />
      );
    };

    const elem = render(<StateWrapper />);

    const search = elem.getByPlaceholderText(placeholder);

    fireEvent.change(search, { target: { value: newValue } });

    expect((search as HTMLInputElement).value).toBe(newValue);
  });
});
