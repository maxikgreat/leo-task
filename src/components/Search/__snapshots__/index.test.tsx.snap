// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component should render correctly and match snapshot 1`] = `
<input
  placeholder="Search by user name..."
  value="Example value"
/>
`;
